src/python/pants/backend/project_info/dependencies.py

```diff
@@ -76,6 +145,25 @@ async def dependencies(
         for address in sorted(address_strings):
             print_stdout(address)
 
+
+@goal_rule
+async def dependencies(
+    console: Console, addresses: Addresses, dependencies_subsystem: DependenciesSubsystem
+) -> Dependencies:
+    if dependencies_subsystem.format == DependenciesOutputFormat.merged:
+        await list_dependencies_as_merged(
+            addresses=addresses,
+            dependencies_subsystem=dependencies_subsystem,
+            console=console,
+        )
+
+    elif dependencies_subsystem.format == DependenciesOutputFormat.json:
```

Comment:
We may add other formats such as DOT or XML.

Reply:
just a stylistic note: I find it more readable to put the differentiator first in the expression, that way scanning the if-elif chain you don't have to jump all the way to the right for each line:
```suggestion
    elif DependenciesOutputFormat.json == dependencies_subsystem.format:
```

Reply:
Sure thing, I've changed the order.